server/service/src/sync/test/merge_helpers.rs

```diff
@@ -0,0 +1,26 @@
+use repository::{
+    mock::{mock_name_link_from_name, MockDataCollection},
+    NameLinkRowRepository, NameRow, RepositoryError, StorageConnection,
+};
+
+pub fn name_links_merged(
```

Comment:
Its because you need a `name_links_merged`, right? I would prefer to be consitend with item_links_merged(), i.e. add some similar option for the mock data. @Chris-Petty any preferences on which version to use? Like Roxies version because its more explicit...

However, the function name could be improved and the function could do with some docs about what it does and what it is used for...

Reply:
Hmm I think I prefer Roxy's, in particular because it doesn't require a silly comment explaining why it isn't part of "all()": 

https://github.com/msupply-foundation/open-msupply/blob/2c21a6b2e282fce3a59eace90cb8c8dcce40407a/server/repository/src/mock/mod.rs#L290

TLDR: The above - do it Roxy's way 😆 

And because... it's more transparently imperative code. Bit of a refactor tangent but all the mock code goes into the mock data inserty thingy which has helper functions for every property that is all boiler plate, but ultimately I don't see the benefit of:

```rust
MockDataInserts::none()
                .mock_units();
                .items()
                .item_links_merged()
                .names()
                .stores()
                .locations()
                .barcodes()
                .full_invoices(),
```
over
```rust
mock_units();
mock_items();
mock_item_links_merged();
mock_names();
mock_stores();
mock_locations();
mock_barcodes();
mock_full_invoices();
`

Where each of those functions just do the upsert on their respective repo.  Ideally each in their own file with the mock data they'll insert. That way when I need to know what data is actually inserted I can go to def and it actually tells me lol. I went in so many circles the other day trying to find where the actual data of `.full_requisitions()` is defined by navigating code, I gave up and just let it run and checked the sqlite db for what data it makes instead lol.

Functionally all I see is it being used in this way, other than `MockDataInserts::all()`, which can just be a function that calls all the other mock functions. Unless it provided something like sync where it ensured things were inserted in referentially intact order all I've felt is pain IMO 😅.

Or if Repos all had a "Repo" trait that ensured `upsert_one` etc. was defined and all data type structs had a trait that ensured you can access their corresponding repo. Then MockDataCollection could generically iterate all the mock data in it (in referential order) but pretty sure this is getting overcooked for tests and really someone will one day want to force in data that is bad for testing IDK 🤪.

Reply:
Good point, also find it hard to follow what has been inserted. However, the sqlite test caching might break...